File: src/app/models/historique-annonce.ts
import { Annonce, AnnonceDto } from './annonce';

export class HistoriqueAnnonce {
  id: number;
  action: string;
  createdDate: Date;

  annonce: Annonce;

}

export class HistoriqueAnnonceDto {
  id: number;
  action: string;
  createdDate: Date;

  annonceDto: AnnonceDto;
}

File: src/main/java/com/chauffeur/models/HistoriqueAnnonce.java
package com.chauffeur.models;

import java.io.Serializable;
import java.util.Date;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.Table;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "historiqueAnnonce")
@Data
@AllArgsConstructor
@NoArgsConstructor
public class HistoriqueAnnonce implements Serializable  {
	
	/**
	 * 
	 */
	private static final long serialVersionUID = 1L;

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long id;

	private Date createdDate;

	private String action;

	@ManyToOne
	@JoinColumn(name = "annonceId")
	private Annonce annonce;

}
